app/controllers/works_controller.rb

```diff
@@ -0,0 +1,91 @@
+class WorksController < ApplicationController
+  before_action :works_all, only: [:index, :main]
+  before_action :find_work, only: [:show, :edit, :update]
+
+  def index; end
+
+  def main; end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+
+    if @work.save
+      flash[:success] = "Successfully created #{@work.category} #{@work.id}"
+      redirect_to work_path(@work)
+    else
+      render :new
+    end
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    @work.assign_attributes(work_params)
+
+    if @work.save
+      flash[:success] = "Successfully updated #{@work.category} #{@work.id}"
+      redirect_to work_path(@work)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @work = Work.find(params[:id])
```

Comment:
This would be a good place to use `find_by` and then ensure that the work was found before continuing on. This will cause issues right now if trying to delete a work that doesn't exist.